### PR TITLE
Create Passkey NPM Project

### LINKED ISSUE #282: Setup New NPM Package for Passkey Contracts
This issue captures the work for setting up a new NPM workspace package for the Passkey contracts.

We decided to use the `safe-modules` repository, especially given how related passkey support is to the ERC-4337 module. This makes this repository more of a collection of extensions to the Safe leveraging its _modularity_ and just "modules" in the strictest terms.

The new package should include basic CI items:
- Empty `hardhat` setup
- Formatting and linting
- Testing (although, this can be a no-op at first)

### PR DESCRIPTION
**Pull request by nlordell**
Fixes #282 

This PR adds an empty NPM package for the Passkey signer "module", including:

- CI configuration
- Default Hardhat template
- `hardhat-deploy` configuration
- Linting

### CODE REVIEW

File: .github/workflows/ci_passkey.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: safe-modules-passkey
+on: [push]
```
Comment (mmv08) [inline .github/workflows/ci_passkey.yml:2]: I'd use the paths filter to run it only when this actual module is changed: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

File: modules/passkey/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@safe-global/safe-passkey",
+  "version": "0.1.0",
+  "author": "@safe-global",
+  "description": "Safe Passkey Owner",
+  "homepage": "https://github.com/safe-global/safe-modules/4337",
```
Comment (mmv08) [inline modules/passkey/package.json:6]: I'd just link the safe-modules repo without nested paths

**Review by mmv08 — APPROVED** [review]

File: modules/passkey/contracts/Lock.sol [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+// Uncomment this line to use console.log
+// import "hardhat/console.sol";
```
[the remark this answers is not in the document]
Comment (nlordell) [reply modules/passkey/contracts/Lock.sol:5]: This was just the default template code from `hardhat init`. I remove the whole file in #298.

File: modules/passkey/package.json (hunk 2 above)
Comment (nlordell) [reply modules/passkey/package.json]: Personally, I like linking to the subdirectory, it makes navigation from NPM more seamless.

Will just fix the path though.

File: .github/workflows/ci_passkey.yml (hunk 1 above)
Comment (nlordell) [reply .github/workflows/ci_passkey.yml]: Added paths filters to CI jobs.

File: .github/workflows/ci_passkey.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+jobs:
+  tests:
```
[the remark this answers is not in the document]
Comment (nlordell) [reply .github/workflows/ci_passkey.yml:8]: To me the job is still "tests", the fact that we generate the coverage report is orthogonal to the purpose of the CI job (which is to ensure testing works).

**Pull request merged**